backend/apis/clients/client-logs/patch-activity-log.api.js

```diff
@@ -0,0 +1,95 @@
+const { app, pool, invalidRequest, databaseError } = require("../../../server");
+const {
+  checkValid,
+  validId,
+  nullableNonEmptyString
+} = require("../../utils/validation-utils");
+const mysql = require("mysql");
+const { modifiableLogTypes } = require("./activity-log.utils");
+
+app.patch(`/api/clients/:clientId/logs/:logId`, (req, res) => {
+  const validationErrors = [
+    ...checkValid(req.params, validId("clientId"), validId("logId")),
+    ...checkValid(
+      req.body,
+      nullableNonEmptyString("title"),
+      nullableNonEmptyString("description")
+    )
+  ];
+
+  if (req.body.logType) {
+    validationErrors.push("logType may not be modified");
+  }
+
+  if (validationErrors.length) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  const selectSql = mysql.format(
+    `SELECT * FROM clientLogs WHERE clientId = ? AND id = ?`,
+    [req.params.clientId, req.params.logId]
+  );
+
+  pool.query(selectSql, (err, result) => {
+    if (err) {
+      return databaseError(req, res, err);
+    }
+
+    if (result.length === 0) {
+      return invalidRequest(
+        res,
+        `No client log exists for client ${req.params.clientId} and log ${req.params.logId}`
+      );
+    } else {
+      const existingLog = result[0];
+
+      if (existingLog.addedBy !== req.session.passport.user.id) {
+        return invalidRequest(
+          res,
+          `You may not modify a client log that you did not create`
```

Comment:
Do you think this could cause issues for if a user is no longer with the organization? Don't know if its not something we need to worry about now.

Reply:
yeah i figure there might come a day when we need "admin" users who can do anything. But that we can build that when the time comes.